mypy/test/dumpmodule.py

```diff
@@ -0,0 +1,148 @@
+import sys
+import types
+from typing import Text
+
+
+if sys.version_info >= (3, 0):
+    long = int
+
+
+def dump_module(id):
+    m = importlib.import_module(id)
+    data = module_to_json(m)
+    print(json.dumps(data, ensure_ascii=True, indent=4, sort_keys=True))
+
+
+def module_to_json(m):
+    result = {}
+    for name, value in m.__dict__.items():
+        # Filter out some useless attributes.
+        if name in ('__file__',
+                    '__doc__',
+                    '__name__',
+                    '__builtins__',
+                    '__package__'):
+            continue
+        if name == '__all__':
+            result[name] = sorted(value)
+        else:
+            result[name] = dump_value(value)
+    return result
+
+
+def dump_value(value, depth=0):
+    if depth > 10:
+        return 'max_recursion_depth_exceeded'
+    if isinstance(value, type):
+        return dump_class(value, depth + 1)
+    if inspect.isfunction(value):
+        return dump_function(value)
+    if callable(value):
+        return 'callable'  # TODO more information
+    if isinstance(value, types.ModuleType):
+        return 'module'  # TODO module name
+    if inspect.isdatadescriptor(value):
+        return 'datadescriptor'
+
+    if inspect.ismemberdescriptor(value):
+        return 'memberdescriptor'
+    return dump_simple(value)
+
+
+def dump_simple(value):
+    if type(value) in (int, bool, float, str, bytes, Text, long, list, set, dict, tuple):
+        return type(value).__name__
+    if value is None:
+        return 'None'
+    if value is inspect.Parameter.empty:
+        return None
+    return 'unknown'
+
+
+def dump_class(value, depth):
+    return {
+        'type': 'class',
+        'attributes': dump_attrs(value, depth),
+    }
+
+
+special_methods = [
+    '__init__',
+    '__str__',
+    '__int__',
+    '__float__',
+    '__bool__',
+    '__contains__',
+    '__iter__',
+]
+
+
+def dump_attrs(d, depth):
+    result = []
+    seen = set()
+    try:
+        mro = d.mro()
+    except TypeError:
+        mro = [d]
+    for base in mro:
+        v = vars(base)
+        for name, value in v.items():
+            if name not in seen:
+                result.append([name, dump_value(value, depth + 1)])
+                seen.add(name)
+    for m in special_methods:
+        if hasattr(d, m) and not m in seen:
+            result.append([m, dump_value(getattr(d, m), depth + 1)])
+    return result
+
+
+kind_map = {
+    inspect.Parameter.POSITIONAL_ONLY: 'POS_ONLY',
+    inspect.Parameter.POSITIONAL_OR_KEYWORD: 'POS_OR_KW',
+    inspect.Parameter.VAR_POSITIONAL: 'VAR_POS',
+    inspect.Parameter.KEYWORD_ONLY: 'KW_ONLY',
+    inspect.Parameter.VAR_KEYWORD: 'VAR_KW',
+}
+
+
+def param_kind(p):
+    s = kind_map[p.kind]
+    if p.default != inspect.Parameter.empty:
+        assert s in ('POS_ONLY', 'POS_OR_KW', 'KW_ONLY')
+        s += '_OPT'
+    return s
+
+
+def dump_function(value):
+    try:
+        sig = inspect.signature(value)
```

Comment:
`inspect.signature` doesn't exist in Python 2. I put together a backport recently (https://github.com/JelleZijlstra/inspect2) and there's a more well-established one at https://pypi.python.org/pypi/funcsigs.